### PR TITLE
docs: fix glob pattern docs

### LINKED ISSUE #387: 📚 The layout of "Glob syntax explained" is disorganized and probably wrong
### Documentation URL

https://biomejs.dev/guides/how-biome-works/#glob-syntax-explained

### Description

Looks like the layout is somehow messed up and IIRC we don't use `globset` in our codebase, so refering to `globset` is probably wrong.

### Expectations

The content should be correct and organized properly.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

Fix #387

### CODE REVIEW

File: src/content/docs/guides/how-biome-works.mdx [hunk 1]
```diff
@@ -335,19 +335,26 @@ This is a current limitation that we will fix in a future release.
...
-> all directories. For example, <code>\*\*&#x2F;foo</code> matches `foo`
-> and `bar/foo` but not `foo/bar`. Secondly, if the glob ends with
-<code>&#x2F;\*\*</code>, then it matches all sub-entries. For example,
-  > <code>foo&#x2F;\*\*</code> matches `foo/a` and `foo/a/b`, but not `foo`.
-  > Thirdly, if the glob contains <code>&#x2F;\*\*&#x2F;</code> anywhere within
-  > the pattern, then it matches zero or more directories. Using `**` anywhere
-  > else is illegal (N.B. the glob `**` is allowed and means "match everything").- Metacharacters such as `*` and `?` can be escaped with character class
-> notation. e.g., `[*]` matches `*`.
+- `dist/**` matches the dist directory and all files in this directory.
+- `**/test/**` matches all files under a directory named `test`.
```
Comment (ematipico) [inline src/content/docs/guides/how-biome-works.mdx:351]:
```suggestion
- `**/test/**` matches all files under any directory named `test`, regardless of where they are. E.g. `dist/test`, `src/test`
```

File: src/content/docs/guides/how-biome-works.mdx [hunk 2]
```diff
@@ -335,19 +335,26 @@ This is a current limitation that we will fix in a future release.
...
+- `**/*.js` matches all files ending with the extension `.js`
```
Comment (ematipico) [inline src/content/docs/guides/how-biome-works.mdx:352]:
```suggestion
- `**/*.js` matches all files in all directories ending with the extension `.js`
```

**Review by ematipico — COMMENTED** [review]
Left a couple of suggestions, let me know what you think

File: src/content/docs/guides/how-biome-works.mdx [hunk 3]
```diff
@@ -335,19 +335,26 @@ This is a current limitation that we will fix in a future release.
...
+A glob pattern specifies a set of filenames.
+Biome supports the following globs:
 
-The syntax and meaning of these two options loosely follow the [globset rules](https://docs.rs/globset/latest/globset/#syntax) but without the ability to set options.
+- `*` matches zero or more characters. It cannot match the path separator `/`.
+- `**` recursively matches directories and files.
+- `?` matches any character. It cannot match the path separator `/`.
+- `[ab]` matches any character among `a` and `b`.
+- `[a-z]` matches any character between `a` and `z`.
+- `[^a-z]` matches any character which is not between `a` and `z`.
```
Comment (Sec-ant) [inline src/content/docs/guides/how-biome-works.mdx:346]: I don't think we support this pattern?
https://github.com/biomejs/biome/blob/a51ef9d771c526c5df42da0167a65813b7e80854/crates/biome_service/src/matcher/pattern.rs#L56-L57

Maybe we should use the docs here: https://github.com/biomejs/biome/blob/a51ef9d771c526c5df42da0167a65813b7e80854/crates/biome_service/src/matcher/pattern.rs#L40-L64

Comment (Conaclos) [reply src/content/docs/guides/how-biome-works.mdx]: Thanks!

File: src/content/docs/guides/how-biome-works.mdx [hunk 4]
```diff
@@ -335,19 +335,29 @@ This is a current limitation that we will fix in a future release.
...
-The syntax and meaning of these two options loosely follow the [globset rules](https://docs.rs/globset/latest/globset/#syntax) but without the ability to set options.
+- `*` matches zero or more characters. It cannot match the path separator `/`.
+- `**` recursively matches directories and files.
+  This sequence **must** form a single path component,
+  so both `**a` and `b**` are invalid and will result in an error.
+  A sequence of more than two consecutive `*` characters is also invalid.
+- `[...]` matches any character inside the brackets.
+  Ranges of characters can also be specified, as ordered by Unicode, so e.g.
+   `[0-9]` specifies any character between 0 and 9 inclusive.
+ - `[!...]` is the negation of `[...]`, i.e. it matches any characters **not** in the brackets.
```
Comment (Sec-ant) [inline src/content/docs/guides/how-biome-works.mdx:349]: These two lines as extra leading spaces.

**Review by Sec-ant — APPROVED** [review]

**Pull request merged**